src/runtime_src/ert/scheduler/scheduler.cpp

```diff
@@ -1097,6 +1096,32 @@ scheduler_loop()
 {
   ERT_DEBUG("ERT scheduler\n");
 
+  /* In Subsytem 2.0, ERT MB now has to go around to access 3 peripherals
+   * internal to ERT Subsystem i.e CQRAM Controller, Embedded scheduler HW
+   * and KDMA. ERT MB needs to read the value in ERT_BASE_ADDR, add that
+   * value to ERT Subsystem Base Address and Peripheral Address, new value
+   * would be used by the ERT MB to access CQ and CSR.
+   */
+  #if defined(ERT_BUILD_V20)
+    ert_base_addr = read_reg(ERT_BASE_ADDR);
+    STATUS_REGISTER_ADDR[0] = ERT_STATUS_REGISTER_ADDR0;
+    STATUS_REGISTER_ADDR[1] = ERT_STATUS_REGISTER_ADDR1;
+    STATUS_REGISTER_ADDR[2] = ERT_STATUS_REGISTER_ADDR2;
+    STATUS_REGISTER_ADDR[3] = ERT_STATUS_REGISTER_ADDR3;
+    CU_DMA_REGISTER_ADDR[0] = ERT_CU_DMA_REGISTER_ADDR0;
+    CU_DMA_REGISTER_ADDR[1] = ERT_CU_DMA_REGISTER_ADDR1;
+    CU_DMA_REGISTER_ADDR[2] = ERT_CU_DMA_REGISTER_ADDR2;
+    CU_DMA_REGISTER_ADDR[3] = ERT_CU_DMA_REGISTER_ADDR3;
+    CU_STATUS_REGISTER_ADDR[0] = ERT_CU_STATUS_REGISTER_ADDR0;
+    CU_STATUS_REGISTER_ADDR[1] = ERT_CU_STATUS_REGISTER_ADDR1;
+    CU_STATUS_REGISTER_ADDR[2] = ERT_CU_STATUS_REGISTER_ADDR2;
+    CU_STATUS_REGISTER_ADDR[3] = ERT_CU_STATUS_REGISTER_ADDR3;
+    CQ_STATUS_REGISTER_ADDR[0] = ERT_CQ_STATUS_REGISTER_ADDR0;
+    CQ_STATUS_REGISTER_ADDR[1] = ERT_CQ_STATUS_REGISTER_ADDR1;
+    CQ_STATUS_REGISTER_ADDR[2] = ERT_CQ_STATUS_REGISTER_ADDR2;
+    CQ_STATUS_REGISTER_ADDR[3] = ERT_CQ_STATUS_REGISTER_ADDR3;
+  #endif
+
```

Comment:
Do this in setup() instead, or since setup() is called also via configuration, maybe instead add a setup_ert_base_addr() function with this code so as to call it only once.   My main objection is polluting scheduler_loop() with this setup specific code.

Reply:
done.